pulsar/src/main/java/com/flipkart/varadhi/pulsar/entities/PulsarProducerResult.java

```diff
@@ -1,11 +1,32 @@
 package com.flipkart.varadhi.pulsar.entities;
 
 import com.flipkart.varadhi.entities.ProducerResult;
+import com.flipkart.varadhi.exceptions.ArgumentException;
+import org.apache.pulsar.client.api.MessageId;
 
-public class PulsarProducerResult extends ProducerResult {
+public class PulsarProducerResult implements ProducerResult {
+    private final MessageId messageId;
+
+    public PulsarProducerResult(MessageId messageId) {
+        this.messageId = messageId;
+    }
 
     @Override
     public int compareTo(ProducerResult o) {
-        return 0;
+        if (null == o) {
+            throw new ArgumentException("Can not compare null ProducerResult.");
+        }
+        if (o instanceof PulsarProducerResult) {
+            return messageId.compareTo(((PulsarProducerResult) o).messageId);
+        } else {
+            throw new ArgumentException(String.format(
+                    "Can not compare different ProducerResult types. Expected ProducerResult is %s, given  %s.",
+                    PulsarProducerResult.class.getName(), o.getClass().getName()
+            ));
+        }
```

Comment:
auto-generated?
Also, there is no requirement for ProduceResult to be comparable. Is there?

Reply:
U mean ProduceResult or ProducerResult ?
For former comparable is not required as it is more then message offset.  Renamed this class to be Offset and PulsarOffset, however retained the Comparable for Offset.